Modules/Core/Common/test/itkImageRegionGTest.cxx

```diff
@@ -0,0 +1,90 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// First include the header file to be tested:
+#include "itkImageRegion.h"
+#include "itkIndexRange.h"
+#include <gtest/gtest.h>
+#include <algorithm>   // For count.
+#include <type_traits> // For remove_const_t and remove_reference_t.
+
+
+// Tests when zero-sized regions are considered to be inside of a region.
+TEST(ImageRegion, ZeroSizedRegionIsInside)
+{
+  const auto check = [](const auto & region) {
+    using RegionType = std::remove_const_t<std::remove_reference_t<decltype(region)>>;
+    using SizeType = typename RegionType::SizeType;
+
+    auto paddedRegion = region;
+    paddedRegion.PadByRadius(1);
+
+    for (const auto & index : itk::ImageRegionIndexRange<RegionType::ImageDimension>(paddedRegion))
+    {
+      const RegionType zeroSizedRegion(index, SizeType{ { 0 } });
+
+      const auto offset = index - region.GetIndex();
+
+      if (std::count(offset.begin(), offset.end(), 0) == 0)
```

Comment:
I am fine with eliminating this exception. I am certain this was not intentional. At first I didn't even understand where it came from 😄

Reply:
@dzenanz My intention is to first have a unit test that describes the original behavior, including the exceptional border cases, and then, when it has passed on the CI, propose the adjustment I had in mind to _both_ the implementation and the test. Please wait...  😃